pubsub/api/PubsubTest.cs

```diff
@@ -0,0 +1,410 @@
+            return subscriber;
+        }
+
+        public void CreateTopic(string topicId, PublisherClient publisher)
+        {
+            // [START create_topic]
+            string topicName = PublisherClient.FormatTopicName(_projectId, 
+                topicId);
+            publisher.CreateTopic(topicName);
+            // [END create_topic]
+        }
+
+        public void CreateSubscription(string topicId, string subscriptionId, 
+            SubscriberClient subscriber)
+        {
+            // [START create_subscription]
+            string topicName = PublisherClient.FormatTopicName(_projectId, 
+                topicId);
+            string subscriptionName = 
+                SubscriberClient.FormatSubscriptionName(_projectId, 
+                subscriptionId);
+            Subscription subscription = subscriber.CreateSubscription(
+                subscriptionName, topicName, pushConfig: null, 
+                ackDeadlineSeconds: 60);
+            // [END create_subscription]
+        }
+
+        public void CreateTopicMessage(string topicId, PublisherClient publisher)
+        {
+            // [START publish_message]
+            string topicName = PublisherClient.FormatTopicName(_projectId, 
+                topicId);
+            PubsubMessage message = new PubsubMessage
+            {
+                // The data is any arbitrary ByteString. Here, we're using text.
+                Data = ByteString.CopyFromUtf8("Hello Cloud Pub/Sub!"),
+                // The attributes provide metadata in a string-to-string 
+                // dictionary.
+                Attributes =
+                {
+                    { "description", "Simple text message" }
+                }
+            };
+            publisher.Publish(topicName, new[] { message });
+            // [END publish_message]
+        }
+
+        public PullResponse PullTopicMessage(string subscriptionId, 
+            SubscriberClient subscriber)
+        {
+            // [START pull_messages]
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, 
+                subscriptionId);
+            PullResponse response = subscriber.Pull(subscriptionName, 
+                returnImmediately: true, maxMessages: 10);
+            // [END pull_messages]
+            return response;
+        }
+
+        public void AcknowledgeTopicMessage(
+            string subscriptionId, 
+            SubscriberClient subscriber, 
+            PullResponse response)
+        {
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, 
+                subscriptionId);
+            // [START pull_messages]
+            subscriber.Acknowledge(subscriptionName, 
+                response.ReceivedMessages.Select(m => m.AckId));
+            // [END pull_messages]
+        }
+
+        public Topic GetTopic(string topicId, PublisherClient publisher)

```

Comment:
GetTopic() and GetSubscription() each have only one call site.  Will later samples invoke them too?